### DISCUSSION TITLE
stargate healthcheck

### DISCUSSION THREAD

**Discussion by deancarrolldev** [stargate/stargate#2056] [Ideas]
Hello, 
We're looking for the best practice for healthchecking astra with the stargate api.

Ideally the health check would also validate that the credentials are correct rather than just a simple okay or now.

In other applications we've used a healthcheck table to which we check if we can periodically write/read to a table.  However after a discussion previously with Datastax they've suggested there may be better alternatives through exposing some internal health checks that are available.

Many Thanks

**Comment by ivansenic** [comment]
I like the idea, especially the part with the credentials. However, at the moment I don't have an idea how would we implement this. Also credentials do have different access scopes, so creating a generic solution would not be possible maybe.

But we should definitely keep an eye on this and try to define some approach that would work..

@jeffreyscarpenter @tatu-at-datastax your opinion?

**Reply by tatu-at-datastax** [reply]
@ivansenic We talked a bit about this but I don't think we came to any conclusion. I also liked the idea wrt customers having some access for health/status of the system; I think Jim from cndb had some concerns (but he may have misunderstood scope). I actually missed the part about credentials checks -- I assumed this was more about availability (and perhaps version) of Stargate, and credentials might not even be needed.

@deancarrollsky what specifically do you mean by credentials validation over and beyond them being ok? To check if they can be used for specific kind of access or ... ?

**Reply by deancarrolldev** [reply]
@tatu-at-datastax It's to catch misconfiguration easier,  them being 'ok' is probably enough.   It would be even better if we could tell that we have permission to write and/or permission to read but we can live without that.

**Reply by tatu-at-datastax** [reply]
@deancarrollsky Ok but misconfiguration of which credentials?

**Comment by jeffreyscarpenter** [comment]
IMO this is an Astra feature request and the OSS project is the wrong place for this discussion.
